packages/core/echo/echo-pipeline/src/space/space.ts

```diff
@@ -210,21 +210,21 @@ export class Space extends Resource {
   }
 }
 
+const SPACE_IDS_CACHE = new ComplexMap<PublicKey, SpaceId>(PublicKey.hash);
+
 /**
  * Space keys are generated by creating a keypair, and then taking the first 20 bytes of the SHA-256 hash of the public key and encoding them to multibase RFC4648 base-32 format (prefixed with B, see Multibase Table).
  * Inspired by how ethereum addresses are derived.
  */
 export const createIdFromSpaceKey = async (spaceKey: PublicKey): Promise<SpaceId> => {
-  // Sanity check.
-  invariant(spaceKey.length === EXPECTED_SPACE_KEY_LENGTH);
+  if (SPACE_IDS_CACHE.has(spaceKey)) {
+    return SPACE_IDS_CACHE.get(spaceKey)!;
+  }
```

Comment:
```suggestion
  const cachedValue = SPACE_IDS_CACHE.get(spaceKey);
  if (cachedValue !== undefined) {
    return cachedValue;
  }
```

this does 2 map lookups instead of 1